lib/common/mainloop.c

```diff
@@ -449,6 +449,56 @@ gio_poll_destroy(gpointer data)
     }
 }
 
+/*!
+ * \internal
+ * \brief Convert libqb's poll priority into GLib's one
+ *
+ * \param[in] prio  libqb's poll priority
+ *
+ * \return  Best matching GLib's priority
+ */
+static gint
+conv_prio_libqb2glib(enum qb_loop_priority prio) {
+        gint ret = G_PRIORITY_DEFAULT;
+        switch (prio) {
+                case QB_LOOP_LOW:
+                        ret = G_PRIORITY_LOW;
+                        break;
+                case QB_LOOP_HIGH:
+                        ret = G_PRIORITY_HIGH;
+                        break;
+                default:
+                        ret = G_PRIORITY_DEFAULT;
+                        break;
+        }
+        return ret;
+}
+
+/*!
+ * \internal
+ * \brief Convert libqb's poll priority to rate limiting spec
+ *
+ * \param[in] prio  libqb's poll priority
+ *
+ * \return  Best matchingrate limiting spec
+ */
+static enum qb_ipcs_rate_limit
+conv_libqb_prio2ratelimit(enum qb_loop_priority prio) {
+        enum qb_ipcs_rate_limit ret = G_PRIORITY_DEFAULT;
```

Comment:
re initialization: looks like an overlook

Reply:
Yes, the point is to propagate the priority to glib, which due to the arrangment
is done indirectly through data carried on libqb's side.

Reply:
Hmm, it was just a miscompile of the test mock based daemon that made
me think the fix won't help -- fortunately the opposite is true :-)

Let me put that test daemon into shape so that the proper reproducer
can be stated in the commit message.

Reply:
This initialization still needs a QB_IPCS_RATE_* constant

Reply:
Dammit, thanks.

Merely out of my curiosity, will try to see if this influence OK-ness with
unpatched libqb (I suppose not) -- but that doesn't block anything here.

Reply:
Nope, that didn't help at all (which would alleviate the need for
updated libqb otherwise) ... allegedly successful run-through before
hence remains a mystery of the lost memories.